azurerm/resource_arm_app_service_virtual_network_association.go

```diff
@@ -0,0 +1,190 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/web/mgmt/2018-02-01/web"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/locks"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmAppServiceVirtualNetworkAssociation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAppServiceVirtualNetworkAssociationCreateUpdate,
+		Read:   resourceArmAppServiceVirtualNetworkAssociationRead,
+		Update: resourceArmAppServiceVirtualNetworkAssociationCreateUpdate,
+		Delete: resourceArmAppServiceVirtualNetworkAssociationDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"app_service_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+			"subnet_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     false,
+				ValidateFunc: azure.ValidateResourceID,
+			},
+		},
+	}
+}
+
+func resourceArmAppServiceVirtualNetworkAssociationCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).web.AppServicesClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Get("app_service_id").(string))
+	if err != nil {
+		return fmt.Errorf("Error parsing Azure Resource ID %q", id)
+	}
+	subnetID, err := parseAzureResourceID(d.Get("subnet_id").(string))
+	if err != nil {
+		return fmt.Errorf("Error parsing Azure Resource ID %q", id)
+	}
+	resourceGroup := d.Get("resource_group_name").(string)
+	name := id.Path["sites"]
+	subnetName := subnetID.Path["subnets"]
+	virtualNetworkName := subnetID.Path["virtualNetworks"]
+
+	locks.ByName(virtualNetworkName, virtualNetworkResourceName)
+	defer locks.UnlockByName(virtualNetworkName, virtualNetworkResourceName)
+
+	locks.ByName(subnetName, subnetResourceName)
+	defer locks.UnlockByName(subnetName, subnetResourceName)
+
+	exists, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(exists.Response) {
+			return fmt.Errorf("Error retrieving existing App Service %q (Resource Group %q): App Service not found in resource group", name, resourceGroup)
+		}
+		return fmt.Errorf("Error retrieving existing App Service %q (Resource Group %q): %s", name, resourceGroup, err)
+	}
+
+	connectionEnvelope := web.SwiftVirtualNetwork{
+		SwiftVirtualNetworkProperties: &web.SwiftVirtualNetworkProperties{
+			SubnetResourceID: utils.String(d.Get("subnet_id").(string)),
+		},
+	}
+	_, err = client.CreateOrUpdateSwiftVirtualNetworkConnection(ctx, resourceGroup, name, connectionEnvelope)
+	if err != nil {
+		return err
```

Comment:
could we wrap this error to make this clearer what's failed?

```suggestion
		return fmt.Errorf("Error creating/updating Association between App Service %q and Virtual Network %q: %s", name, subnetID)
```